Assets/Scripts/Game/Core/Gameplay/GameEvents/EventPanel/EventPanelContext/EventPaneWorkerContractFinishedContext.cs

```
@@ -34,4 +50,17 @@ public string GetTitle()
     private void CloseButton()
     {
     }
+
+    private string GetEventLine(CompleteWorkerContractEvent completeWorkerContract)
+    {
+        GameFlowService gameFlowService = ServiceLocator.Instance.Get<GameFlowService>();
+        PlayersService playersService = ServiceLocator.Instance.Get<PlayersService>();
+
+        IWorker worker = completeWorkerContract.Worker;
+        string employerName = playersService.GetPlayerName(completeWorkerContract.Employer);
+        bool workerIsCitizen = (worker.WorkerClass.WorkerClassType == WorkerClassType.Citizen);
+        if (workerIsCitizen)
```

Actually, a slave can never finish their contract :D (instead they are "freed"). So I guess that if this is not a citizen it should raise an error.